install/kubernetes/cilium/templates/clustermesh-apiserver/deployment.yaml

```diff
@@ -48,41 +48,21 @@ spec:
       {{- end }}
       initContainers:
       - name: etcd-init
-        image: {{ include "cilium.image" .Values.clustermesh.apiserver.etcd.image | quote }}
-        imagePullPolicy: {{ .Values.clustermesh.apiserver.etcd.image.pullPolicy }}
-        command: ["/bin/sh", "-c"]
+        image: {{ include "cilium.image" .Values.clustermesh.apiserver.image | quote }}
+        imagePullPolicy: {{ .Values.clustermesh.apiserver.image.pullPolicy }}
         args:
-        - |
-          rm -rf /var/run/etcd/*;
-          /usr/local/bin/etcd --data-dir=/var/run/etcd --name=clustermesh-apiserver --listen-client-urls=http://127.0.0.1:2379 --advertise-client-urls=http://127.0.0.1:2379 --initial-cluster-token=clustermesh-apiserver --initial-cluster-state=new --auto-compaction-retention=1 &
-
-          # The following key needs to be created before that the cilium agents
-          # have the possibility of connecting to etcd.
-          etcdctl put cilium/.has-cluster-config true
-
-          etcdctl user add root --no-password;
-          etcdctl user grant-role root root;
-          etcdctl user add admin-{{ .Values.cluster.name }} --no-password;
-          etcdctl user grant-role admin-{{ .Values.cluster.name }} root;
-          etcdctl user add externalworkload --no-password;
-          etcdctl role add externalworkload;
-          etcdctl role grant-permission externalworkload --from-key read '';
-          etcdctl role grant-permission externalworkload readwrite --prefix cilium/state/noderegister/v1/;
-          etcdctl role grant-permission externalworkload readwrite --prefix cilium/.initlock/;
-          etcdctl user grant-role externalworkload externalworkload;
-          etcdctl user add remote --no-password;
-          etcdctl role add remote;
-          etcdctl role grant-permission remote --from-key read '';
-          etcdctl user grant-role remote remote;
-          etcdctl auth enable;
-          exit
-        env:
-        - name: ETCDCTL_API
-          value: "3"
-        - name: HOSTNAME_IP
-          valueFrom:
-            fieldRef:
-              fieldPath: status.podIP
+          - /usr/bin/clustermesh-apiserver
+        args:
+          - etcdinit
+        {{- if .Values.debug.enabled }}
+          - --debug
+        {{- end }}
+          - --etcd-cluster-name=$(CLUSTER_NAME)
+          - --etcd-initial-cluster-token=clustermesh-apiserver
+          - --etcd-data-dir=/var/run/etcd
+          # This is the Cilium cluster name, it's hardcoded to clustermesh-apiserver for now. This is used to determine
+          # the username of the admin account.
+          - --cluster-name=clustermesh-apiserver
```

Comment:
`--cluster-name` and `--etcd-cluster-name` are swapped (the former should be set to `$(CLUSTER_NAME)`, and the other hardcoded, or completely omitted).

Reply:
Oops 😬